mypy/types.py

```diff
@@ -27,6 +27,19 @@
 
 JsonDict = Dict[str, Any]
 
+# The set of all valid expressions that can currently be contained
+# inside of a Literal[...].
+#
+# Literals can contain enum-values: we special-case those and
+# store the value as a string.
+#
+# Note: this type also happens to correspond to types that can be
+# directly converted into JSON. The serialize/deserialize methods
+# of 'LiteralType' rely on this, as well 'server.astdiff.SnapshotTypeVisitor'
+# and 'types.TypeStrVisitor'. If we end up adding any non-JSON-serializable
+# types to this list, we should make sure to edit those methods to match.
+LiteralInnerExpr = Union[int, str, bool, None]
```

Comment:
Would we need to be able to distinguish between `unicode` and `str` literals in Python 2? For example, the `open()` mode argument could well be a unicode literal if using `from __future__ import unicode_literals`.

A name like `LiteralValue` would perhaps be a bit more descriptive. The current name implies to me that this is an `Expression` object, which is not the case.